### PR TITLE
Social sharing. Filter out bad parts

### PR DESCRIPTION
**Pull request by roman-khomenko-dev**
Check for a bad social_part data and highlight them in UI

### CODE REVIEW

File: lib/chat_web/live/main_live/page/recover_key_share.ex [hunk 1]
```diff
@@ -127,7 +150,7 @@ defmodule ChatWeb.MainLive.Page.RecoverKeyShare do
         value="Accept"
         phx-click="accept"
         phx-target={@myself}
-        disabled={!@changeset.valid?}
+        disabled={!@changeset.valid? || Enum.any?(@shares, &KeyShare.broken?(&1))}
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/page/recover_key_share.ex:153]: `&fun/1 ` syntax is better here

File: lib/chat_web/live/main_live/page/recover_key_share.ex [hunk 2]
```diff
@@ -247,6 +270,13 @@ defmodule ChatWeb.MainLive.Page.RecoverKeyShare do
     """
   end
 
+  defp check_broken_content(socket, shares),
+    do:
+      if(Enum.any?(shares, &KeyShare.broken?(&1)),
```
Comment (sergey-lukianov) [inline lib/chat_web/live/main_live/page/recover_key_share.ex:275]: same

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@ Here we write upgrading notes for brands. It's a team effort to make them as
 straightforward as possible.
 
 ### Added
+- Social sharing. filter out bad parts
```
Comment (sergey-lukianov) [inline CHANGELOG.md:14]: update `main` brach and move this line in unreleased section

**Review by sergey-lukianov — CHANGES_REQUESTED** [review]

**Review by sergey-lukianov — CHANGES_REQUESTED** [review]

File: lib/chat/key_share.ex [hunk 4]
```diff
@@ -75,13 +77,21 @@ defmodule Chat.KeyShare do
...
   def read_content(path) do
-    path
-    |> File.stream!()
-    |> Stream.map(&String.trim_trailing/1)
-    |> Enum.map(&decode_content/1)
+    case path
+         |> File.stream!()
+         |> Stream.map(&String.trim_trailing/1)
+         |> Enum.map(&decode_content/1) do
+      [:error | _] -> :error
```
[the remark this answers is not in the document]
Comment (roman-khomenko-dev) [reply lib/chat/key_share.ex]: changed to use `content_result/1` where check for any element is :error

**Review by sergey-lukianov — APPROVED** [review]

**Pull request merged**